pkgs/applications/version-management/gitkraken/default.nix

```
@@ -127,10 +127,20 @@ let
         patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $file
       done
 
-      for file in $(find . -type f \( -name \*.node -o -name ${pname} -o -name \*.so\* \) ); do
+      for file in $(find . -type f \( -name \*.node -o -name ${pname} -o -name git -o -name git-\* -o -name scalar -o -name \*.so\* \) ); do
         patchelf --set-rpath ${libPath}:$out/share/${pname} $file || true
       done
       popd
+
+      pushd $out/share/${pname}/resources/app.asar.unpacked/node_modules/@axosoft/nodegit/build/Release
+      mv nodegit-ubuntu-18.node nodegit-ubuntu-18-ssl-1.1.1.node
+      mv nodegit-ubuntu-18-ssl-static.node nodegit-ubuntu-18.node
+      chmod 755 nodegit-ubuntu-18.node
+      popd
+
+      rm -rf $out/share/${pname}/resources/app.asar.unpacked/git
+      ln -s ${git} $out/share/${pname}/resources/app.asar.unpacked/git
+      makeWrapper $out/share/${pname}/${pname} $out/bin/${pname} --set GIT_SSL_CAINFO "${cacert}/etc/ssl/certs/ca-bundle.crt"
```

```suggestion
      rm -rf $out/share/gitkraken/resources/app.asar.unpacked/git
      ln -s ${git} $out/share/gitkraken/resources/app.asar.unpacked/git
      makeWrapper $out/share/gitkraken/gitkraken $out/bin/gitkraken --set GIT_SSL_CAINFO "${cacert}/etc/ssl/certs/ca-bundle.crt"
```
using pname as much as possible where the string matches is an anti pattern

---

Just wanted to stick with what was already being done, like on lines 115-116:

```
mkdir -p $out/share/${pname}/
cp -R $src/* $out/share/${pname}
```

Since the `cp` command uses `pname`, shouldn’t we stick to this pattern for paths that reference the `$out/share/${pname}` directory?

However, I would agree to change line 143 since the original `gitkraken` binary is something fixed that we don’t want (nor need) to tinker with:

```diff
-      makeWrapper $out/share/${pname}/${pname} $out/bin/${pname} --set GIT_SSL_CAINFO "${cacert}/etc/ssl/certs/ca-bundle.crt"
+      makeWrapper $out/share/${pname}/gitkraken $out/bin/gitkraken --set GIT_SSL_CAINFO "${cacert}/etc/ssl/certs/ca-bundle.crt"
```

---

Regarding the PR title, would something like `gitkraken: patch bundled git` be more explicit about what this PR does?

---

> Since the `cp` command uses `pname`, shouldn’t we stick to this pattern for paths that reference the `$out/share/${pname}` directory?

Using pname in many places where it is not logically linked and just by coincidence the same string, is bad practice. pname could also be gitkrane-something and then things would break.

> Regarding the PR title, would something like `gitkraken: patch bundled git` be more explicit about what this PR does?

We are not really patching the binary itself either but replace it with the git from nixpkgs, so we basically devendor it.

---

Ok fair enough. Should I update all `pname` usages or just the ones from my addition?

I updated the PR name to better reflect the changes. Thanks for your insight on this.

---

if its not too many changes then I would change all pname references in this PR

---

The current patch I have is the following:

```diff
--- a/pkgs/applications/version-management/gitkraken/default.nix
+++ b/pkgs/applications/version-management/gitkraken/default.nix
@@ -97,9 +97,9 @@ let
     ];
 
     desktopItems = [ (makeDesktopItem {
-      name = pname;
-      exec = pname;
-      icon = pname;
+      name = "GitKraken";
+      exec = "gitkraken";
+      icon = "gitkraken";
       desktopName = "GitKraken";
       genericName = "Git Client";
       categories = [ "Development" ];
@@ -112,35 +112,35 @@ let
     installPhase = ''
       runHook preInstall
 
-      mkdir -p $out/share/${pname}/
-      cp -R $src/* $out/share/${pname}
+      mkdir -p $out/share/gitkraken/
+      cp -R $src/* $out/share/gitkraken
 
       mkdir -p $out/share/pixmaps
-      cp ${pname}.png $out/share/pixmaps/${pname}.png
+      cp gitkraken.png $out/share/pixmaps/gitkraken.png
 
       runHook postInstall
     '';
 
     postFixup = ''
-      pushd $out/share/${pname}
-      for file in ${pname} chrome-sandbox chrome_crashpad_handler; do
+      pushd $out/share/gitkraken
+      for file in gitkraken chrome-sandbox chrome_crashpad_handler; do
         patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $file
       done
 
-      for file in $(find . -type f \( -name \*.node -o -name ${pname} -o -name git -o -name git-\* -o -name scalar -o -name \*.so\* \) ); do
-        patchelf --set-rpath ${libPath}:$out/share/${pname} $file || true
+      for file in $(find . -type f \( -name \*.node -o -name gitkraken -o -name git -o -name git-\* -o -name scalar -o -name \*.so\* \) ); do
+        patchelf --set-rpath ${libPath}:$out/share/gitkraken $file || true
       done
       popd
 
-      pushd $out/share/${pname}/resources/app.asar.unpacked/node_modules/@axosoft/nodegit/build/Release
+      pushd $out/share/gitkraken/resources/app.asar.unpacked/node_modules/@axosoft/nodegit/build/Release
       mv nodegit-ubuntu-18.node nodegit-ubuntu-18-ssl-1.1.1.node
       mv nodegit-ubuntu-18-ssl-static.node nodegit-ubuntu-18.node
       chmod 755 nodegit-ubuntu-18.node
       popd
 
-      rm -rf $out/share/${pname}/resources/app.asar.unpacked/git
-      ln -s ${git} $out/share/${pname}/resources/app.asar.unpacked/git
-      makeWrapper $out/share/${pname}/${pname} $out/bin/${pname} --set GIT_SSL_CAINFO "${cacert}/etc/ssl/certs/ca-bundle.crt"
+      rm -rf $out/share/gitkraken/resources/app.asar.unpacked/git
+      ln -s ${git} $out/share/gitkraken/resources/app.asar.unpacked/git
+      makeWrapper $out/share/gitkraken/gitkraken $out/bin/gitkraken --set GIT_SSL_CAINFO "${cacert}/etc/ssl/certs/ca-bundle.crt"
     '';
   };
```

I’m wondering if I should keep the `$out/share/${pname}`.

---

haha ok, in my humble opinion I wouldn't touch it.  But since you are a package maintainer, why not ?

---

For what it’s worth, I’m running it like this since a whole week.

IMO we should keep the `$out/share${pname}` to avoid some obvious repetitions, which would leave us with the following:

```diff
--- a/pkgs/applications/version-management/gitkraken/default.nix
+++ b/pkgs/applications/version-management/gitkraken/default.nix
@@ -97,9 +97,9 @@ let
     ];
 
     desktopItems = [ (makeDesktopItem {
-      name = pname;
-      exec = pname;
-      icon = pname;
+      name = "GitKraken";
+      exec = "gitkraken";
+      icon = "gitkraken";
       desktopName = "GitKraken";
       genericName = "Git Client";
       categories = [ "Development" ];
@@ -116,18 +116,18 @@ let
       cp -R $src/* $out/share/${pname}
 
       mkdir -p $out/share/pixmaps
-      cp ${pname}.png $out/share/pixmaps/${pname}.png
+      cp gitkraken.png $out/share/pixmaps/gitkraken.png
 
       runHook postInstall
     '';
 
     postFixup = ''
       pushd $out/share/${pname}
-      for file in ${pname} chrome-sandbox chrome_crashpad_handler; do
+      for file in gitkraken chrome-sandbox chrome_crashpad_handler; do
         patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $file
       done
 
-      for file in $(find . -type f \( -name \*.node -o -name ${pname} -o -name git -o -name git-\* -o -name scalar -o -name \*.so\* \) ); do
+      for file in $(find . -type f \( -name \*.node -o -name gitkraken -o -name git -o -name git-\* -o -name scalar -o -name \*.so\* \) ); do
         patchelf --set-rpath ${libPath}:$out/share/${pname} $file || true
       done
       popd
@@ -140,7 +140,7 @@ let
 
       rm -rf $out/share/${pname}/resources/app.asar.unpacked/git
       ln -s ${git} $out/share/${pname}/resources/app.asar.unpacked/git
-      makeWrapper $out/share/${pname}/${pname} $out/bin/${pname} --set GIT_SSL_CAINFO "${cacert}/etc/ssl/certs/ca-bundle.crt"
+      makeWrapper $out/share/${pname}/gitkraken $out/bin/gitkraken --set GIT_SSL_CAINFO "${cacert}/etc/ssl/certs/ca-bundle.crt"
     '';
   };
```

I’m personnally fine with this change :slightly_smiling_face:

---

@SuperSandro2000 I kept usages of `$out/share/${pname}` and replaced all other usages of `pname` with raw strings.